### PR TITLE
OAuth and related changes to project permissions

### PR DESCRIPTION
**Pull request by johnbradley**
Users can login and use their credentials to browse projects/files in DukeDS.
When the user clicks start/restart the bespin user receives download permissions for any files the user has chosen for the job.

### CODE REVIEW

File: data/api.py [hunk 1]
```diff
@@ -149,23 +149,10 @@ class DDSEndpointViewSet(viewsets.ModelViewSet):
...
-    def perform_create(self, serializer):
-        self.save_with_user(serializer)
-
-    def perform_update(self, serializer):
-        self.save_with_user(serializer)
-
-    def save_with_user(self, serializer):
-        serializer.save(user=self.request.user)
+    serializer_class = ReadOnlyDDSUserCredSerializer
+    queryset = DDSUserCredential.objects.all()
```
Comment (dleehr) [inline data/api.py:155]: So the credentials will be readable by any authenticated API user?

**Review by dleehr — COMMENTED** [review]
Looks good. Just that one question about the DDS credentials API. Assuming users have bespin-api credentials, wouldn't this give them access to our service account's DDS token?

File: data/api.py (hunk 1 above)
Comment (johnbradley) [reply data/api.py:155]: The only credentials stored here will be for the bespin user created by an admin user.
They can't see the token just see the name and the id.
Initially there will only be a single credential value for production DukeDS.
Long term they may want to pick from different DukeDS instances.

File: data/serializers.py [hunk 2]
```diff
@@ -98,10 +97,16 @@ class Meta:
         fields = ('id', 'user', 'token', 'endpoint')
 
 
+class ReadOnlyDDSUserCredSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DDSUserCredential
+        resource_name = 'dds-user-credentials'
+        fields = ('id', 'user', 'endpoint')
```
Comment (dleehr) [inline data/serializers.py]: I see, so the `ReadOnlyDDSUserCredSerializer` only serializes the id, user and endpoint fields - not the actual credentials in `DDSUserCredential`

Comment (dleehr) [reply data/serializers.py]: Probably worth a comment here to indicate why this serializer omits key fields (e.g. to suppress our system credentials) and definitely a test case to prove that a standard user does not see these fields through the API.

**Pull request merged**